test/libsolidity/syntaxTests/parsing/arrays_in_expressions.sol

```diff
@@ -1,8 +1,8 @@
 contract c {
-    function f() { c[10] a = 7; uint8[10 * 2] x; }
+    function f() public { c[10] a = 7; uint8[10 * 2] x; }
```

Comment:
I saw this test earlier in another PR - I'm not really sure what it's supposed to test and whether it makes sense to fix it. It's probably supposed to only test parsing of arrays and the thought was probably that for that it doesn't matter, if it fails later, but maybe the test should still not fail like this? Doesn't necessarily need to be done in this PR, though.

Reply:
Indeed, it is just a parser test. We could perhaps add some `memory` or `storage` keywords, especially once this turns into an error.